azurelinuxagent/ga/exthandlers.py

```diff
@@ -278,15 +278,30 @@ def get_goal_state_debug_metadata(self):
         :return: Tuple (activity_id, correlation_id, gs_created_timestamp) or "NA" for any property that's not available
         """
 
-        def check_empty(value): return value if value not in (None, "") else "NA"
+        def parse_value(parse_fn, value):
+
+            try:
+                if value not in (None, ""):
+                    return parse_fn(value)
+            except Exception as e:
+                # A failure here isn't a fatal error, because the info we're
+                # trying to retrieve is debug only on linux.
+                error_msg = u"Couldn't parse debug metadata value: {0}".format(e)
+                logger.verbose(error_msg)
+                return value
```

Comment:
the type of the `in_vm_gs_metadata.created_on_ticks` is datetime so just returning this wont work. If you want to return the string value of what we get from CRP, we need to modify this - `self.created_on_ticks = __ticks_to_datetime(getattrib(in_vm_metadata_node, "createdOnTicks"))`